<a href="https://colab.research.google.com/github/ssnyu/quant-strategies/blob/main/6_Appears_before_5.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


### Roll a fair standard 6-sided die until a 6 appears. Given that the first 6 occurs before the first 5, find the expected number of times the die was rolled

In [26]:
import random

n_simulations = 100000

Average_Count = 0

for i in range(n_simulations):
  six_found = False
  five_occured = False
  count = 0
  while(not six_found):
    dice_throw = random.randint(1,6) #simulating a dice throw
    count+=1
    if dice_throw == 5:
      five_occured = True # check if 5 has occured
      break

    if dice_throw == 6 and not five_occured:
      six_found = True
  Average_Count += count

print(Average_Count/n_simulations)

3.00567


**Conclusions**
<br>
In this simulation, we investigated the expected number of times a fair 6-sided die needs to be rolled until a 6 appears, given that the first 6 occurs before the first 5. Through 100,000 simulations, we obtained an average count of 3.00567, which closely matches the mathematical solution of 6.
This result demonstrates the accuracy of the mathematical approach and the reliability of simulation methods in estimating probabilities. The close agreement between the simulated and mathematical solutions validates the assumptions made in the problem and confirms our understanding of conditional probability and geometric distribution.
In conclusion, this exercise showcases the power of combining mathematical derivations with simulation techniques to solve complex probability problems, providing a comprehensive understanding of the underlying phenomena.